packages/hash/api/src/graphql/typeDefs/ontology/property-type.typedef.ts

```diff
@@ -17,22 +42,47 @@ export const propertyTypeTypedef = gql`
     The property type
     """
     propertyType: PropertyType!
-    # TODO: we might need something like
-    # "referencedDataTypes: [PersistedDataType!]"
-    # for us to retrieve all referenced data types, and one for referenced property types as well.
-    # This will be helpful for displaying the entire property type in the frontend
+    # INTERFACE FIELDS END #
+  }
+
+  type PersistedPropertyTypeWithRefs implements PersistedPropertyTypeInterface {
```

Comment:
The issue is, that we don't want to allow nesting in GQL, otherwise, this would be possible:
```gql
query recursive_property_types {
  getAllLatestPropertyTypes {
    propertyType

    referencedDataTypes {
      dataType
    }

    referencedPropertyTypes {
      referencedPropertyTypes {
        referencedPropertyTypes {
          referencedPropertyTypes {
            propertyType
          }
        }
      }
    }
  }
}
```

The references are only resolved at the top level